Backend/src/main/java/ktsnvt/tim1/services/AuthenticationService.java

```diff
@@ -31,14 +33,21 @@
     private VerificationTokenRepository verificationTokenRepository;
 
     @Autowired
-    private Environment environment;
+    private EmailService emailService;
 
     @Autowired
-    private EmailService emailService;
+    private UserMapper userMapper;
 
+    @Transactional
     public UserDTO register(UserDTO user, HttpServletRequest request) throws EntityAlreadyExistsException, MalformedURLException {
         String url = new URL(request.getRequestURL().toString()).getAuthority();
 
+        if(verificationTokenRepository.findByUser_Email(user.getEmail())!=null && verificationTokenRepository.findByUser_Email(user.getEmail()).isExpired() &&
```

Comment:
This code for deleting expired verification tokens and corresponding users should be in PeriodicCheckService and activate, for example, at 00:00 every day